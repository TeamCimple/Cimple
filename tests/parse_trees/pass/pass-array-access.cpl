int main() {
	int *a;
	
	a = make int[100];
	a[0] = 1;
	return a[0];
}
