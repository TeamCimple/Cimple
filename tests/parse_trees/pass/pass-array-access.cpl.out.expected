FuncDecl(Name: Var(main) ReturnType: int Parameters:  Body: CompoundStatement(Declarations: (int depth: 1 Var(a),),  StatementList: Statement(EQUALS(Identifier(a), Make([int]))), Statement(EQUALS(ArrayAccessIdentifier(a)[Int(0)], Int(1))), RETURN(ArrayAccessIdentifier(a)[Int(0)]))) 
